### PR TITLE
[FIX] calendar : fix the problem of a user making an appointment in a company other than his default company

### PR DESCRIPTION
**Pull request by DjamelTouati**
Steps to follow to reproduce the bug:

- Create a new “Internal User”
- Add him a “default company”
- Add in the “allowed companies” a different company than his default company
- In the “Technical” section, check that the “multi companies” box is checked
- Log in with the new user
- At the top right select a company other than his “default company”
- In the “Calendar” app/ create a new meeting with a future date
- Add another user as a participant
- Confirm the meeting
- An access rights error is triggered

Problem:
When participant are added, an invitation by email must be sent to all participants. The problem is in the template of the email that is sent, precisely in the "body", we add the name of the user's default company rather than the one that is currently set.

Solution:
Use the correct company name in the template and add the translation key in the "calendar.pot".

opw-2494399

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by nle-odoo — COMMENTED** [review]
this will break all translation of all instance that mail.template after the fix, so if it was really necessary (and could not be done in master), the translation should be updated

I don't understand your line:

> The existing users cannot update the "calendar" module to fix the problem. So I added a line that updates the context to have all the authorised companies for the user. In order to be able to read the name of the user default company in the template.
Unfortunately, in the email we will not have the correct user's company name but rather the default one. 

An existing user can just update the view manually if he has the issue. Since this is in noupdate this seems to be the only way to fix it on an existing instance.

**Comment by DjamelTouati** [comment]
Hello @nle-odoo 
Thank you for your review,
I tried to create a temporary solution that fixes the problem for new and existing customers because otherwise other customers who have the same problem will open a new ticket to report the problem. But I think you are right it's better to close this PR and make the change only in the template and create a new PR targeting the master. I'll explain to the customer who reported the issue how he can update the template. What do you think ?

**Comment by nle-odoo** [comment]
Yes, or you can update the customer's template directly.

Usually when breaking translation it depends if it's worth it or not, sometimes it is a big issue enough that breaking the translation is all right (here since this is in noupdate I think translation would not be affected for existing installation of calendar so it would lessen the effect).

Besides that my comment was that I did not think that:

```py
            'base_url': self.env['ir.config_parameter'].sudo().get_param('web.base.url', default='http://localhost:8069'),
            'allowed_company_ids': self.env.user.company_ids.ids, # can be removed in master, see commit message
        })
```

should have changed anything to:

`${object.event_id.user_id.company_id.name}`


in short: user.company_id or user_id.company_id should always give you the value of the field and will not be overridden by allowed_company_ids

`user_id.company_id` should get the exact value of the field without taking allowed_company_ids into any account.

allowed_company_ids should normally only effect env.company as seen in this code:

https://github.com/odoo/odoo/blob/ae16e47944bdfaa5c044b6c76a1102f9f2f525a1/odoo/api.py#L555-L562

and since you set `allowed_company_ids` to `self.env.user.company_ids.ids` the value of env.company will be the first value of self.env.user.company_ids which I think will make it random and not the currently we are currently using.

**Comment by arj-odoo** [comment]
Looks good to me regarding the latests modifications.

**Comment by nle-odoo** [comment]
robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo r-

This breaks translations.

**Comment by tde-banana-odoo** [comment]
Oupsie

@robodoo r+

**Comment by tde-banana-odoo** [comment]
Hello,

Actually I am not sure about your use case, sorry for disturbing again. I prefer to @robodoo r- and then see to r+ again if necessary.

Right company is effectively the one from the event responsible, because he is the one inviting people. If there are access rights issue, maybe some sudo should be applied. It seems strange to me that TdeOdoo invited FheOdoo for a meeting from OdooUS, but as the event inviter company is actually OdooBE.

Could you enlighten me a bit, maybe I did not understand something ?

Many thanks and sorry for r+/r-,

**Comment by tde-banana-odoo** [comment]
@robodoo r- (not sure what he will do with hereabove message)

**Comment by nle-odoo** [comment]
Hi


With the company selector, in many case if you are the user of the event your current company will not match the company on your user.

If

**Comment by nle-odoo** [comment]
As far as I can tell the issue is that: a user (that may be the same as the responsible) invite and the company does not match the current company of the selector.

So the fix was to yes: using current company of the inviter.

If we want to keep company of the user then yes the current commit is not good and it should either be left as is (at least in stable) or maybe (in master?) only use company selector if the current user is the responsible:

```diff
+ % set company = object.env.company if object.event_id.user_id == object.env.user else object.event_id.user_id.company_id
```

**Comment by tde-banana-odoo** [comment]
"If we want to keep company of the event responsible"

As far as I understand this template that's the purpose of this sentence. Otherwise it makes no sense: you are invited to a meeting "from CompanyA", but I tell you it is from CompanyB. This is quite strange.

I would simply strip that company information but some PO wanted to keep it...

**Comment by DjamelTouati** [comment]
Finally, the PO decided to remove the company name information from the email template.

Here is the link to the new PR: https://github.com/odoo/odoo/pull/69935

Thank you for your review!
